lib/queue.rb

```diff
@@ -1,31 +1,65 @@
 class Queue
 
+  MAX_SIZE = 20
+
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(MAX_SIZE)
+    @front = -1
+    @back = -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == 0 && @back == MAX_SIZE - 1 || @back == (@front - 1) % (MAX_SIZE - 1)
+      raise ArgumentError
+    elsif @front == -1
+      @front = 0
+      @back = 0
+    elsif @back == MAX_SIZE - 1 && @front != 0
+      @back = 0
+    else
+      @back += 1
+    end
+    @store[@back] = element
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    return nil if @store.empty?
+    raise ArgumentError if @front == -1
+    element = @store[@front]
+    @store[@front] = nil
+
+    if @front == @back
+      @front = -1
+      @back = -1
+    elsif @front == MAX_SIZE - 1
+      @front = 0
+    else
+      @front += 1
+    end
+    return element
   end
 
   def front
-    raise NotImplementedError, "Not yet implemented"
+    return @store[@front]
   end
 
   def size
-    raise NotImplementedError, "Not yet implemented"
+    return 0 if @front == 1
+    return @front < @back ? @back - @front + 1 : MAX_SIZE - @front + @back
   end
 
   def empty?
-    raise NotImplementedError, "Not yet implemented"
+    @front == -1
   end
 
   def to_s
-    return @store.to_s
+    result_array = []
+    20.times do |i|
```

Comment:
Do you need the magic number `20.times` loop?  Could you instead start `i` at `@front` and keep incrementing until you hit the back?